src/utilities.ts

```diff
@@ -1,9 +1,16 @@
 import axios, { AxiosError } from "axios";
 import { soosLogger } from "./logging/SOOSLogger";
-const isNil = (value: unknown): value is null | undefined => value === null || value === undefined;
+const isNil = (value: unknown): value is null | undefined =>
+  value === null || value === undefined || value === "";
+
+const isNilOrEmpty = (value: unknown): value is null | undefined => {
```

Comment:
this one is only really useful for strings - probably should constrain the type rather than unknown

Reply:
renamed to isEmptyString, used in conjunction with isNil on ensureValue instead of just wrapping it here